### ISSUE TITLE
Deno?

### ISSUE THREAD

**Issue by denizdogan** [probot/smee-client#236]
Has anyone managed to get this working with Deno? I can't seem to figure it out.

```ts
❯ cat foobar.ts
import SmeeClient from "npm:smee-client";

const smee = new SmeeClient({
  source: "https://smee.io/xyzxyzxyzxyz",
  target: "http://localhost:3000/api/webhook",
  logger: console,
});
smee.start();

❯ deno run --unstable -A --check --reload --no-lock foobar.ts
Check file:///Users/deniz/Code/foobar/github-app/foobar.ts
Forwarding https://smee.io/xyzxyzxyzxyz to http://localhost:3000/api/webhook
Connected https://smee.io/xyzxyzxyzxyz
[trigger a webhook]
error: Uncaught (in promise) TypeError: req.setNoDelay is not a function
    at Request.request (file:///Users/deniz/Library/Caches/deno/npm/registry.npmjs.org/superagent/7.1.6/lib/node/index.js:844:7)
    at Request.end (file:///Users/deniz/Library/Caches/deno/npm/registry.npmjs.org/superagent/7.1.6/lib/node/index.js:998:8)
    at Client.onmessage (file:///Users/deniz/Library/Caches/deno/npm/registry.npmjs.org/smee-client/1.2.3/index.js:35:13)
    at EventSource.emit (ext:deno_node/_events.mjs:379:28)
    at _emit (file:///Users/deniz/Library/Caches/deno/npm/registry.npmjs.org/eventsource/1.1.2/lib/eventsource.js:264:17)
    at parseEventStreamLine (file:///Users/deniz/Library/Caches/deno/npm/registry.npmjs.org/eventsource/1.1.2/lib/eventsource.js:279:9)
    at IncomingMessageForClient.<anonymous> (file:///Users/deniz/Library/Caches/deno/npm/registry.npmjs.org/eventsource/1.1.2/lib/eventsource.js:238:11)
    at IncomingMessageForClient.emit (ext:deno_node/_stream.mjs:1852:9)
    at addChunk (ext:deno_node/_stream.mjs:2874:16)
    at readableAddChunk (ext:deno_node/_stream.mjs:2853:13)

    
❯ cat foobar.ts
import SmeeClient from "https://esm.sh/smee-client";

const smee = new SmeeClient({
  source: "https://smee.io/xyzxyzxyzxyz",
  target: "http://localhost:3000/api/webhook",
  logger: console,
});
smee.start();

❯ deno run --unstable -A --check --reload --no-lock foobar.ts
Check file:///Users/deniz/Code/foobar/github-app/foobar.ts
Forwarding https://smee.io/xyzxyzxyzxyz to http://localhost:3000/api/webhook
Connected https://smee.io/xyzxyzxyzxyz
[trigger a webhook]
error: Uncaught Error: Browser-only version of superagent could not find XHR
    at Function.f.getXHR (https://esm.sh/v125/superagent@7.1.5/denonext/superagent.mjs:6:1481)
    at p._end (https://esm.sh/v125/superagent@7.1.5/denonext/superagent.mjs:7:1292)
    at p.end (https://esm.sh/v125/superagent@7.1.5/denonext/superagent.mjs:7:884)
    at f.onmessage (https://esm.sh/v125/smee-client@1.2.3/denonext/smee-client.mjs:2:1940)
    at s.emit (ext:deno_node/_events.mjs:379:28)
    at C (https://esm.sh/v125/eventsource@1.1.2/denonext/eventsource.mjs:2:4000)
    at S (https://esm.sh/v125/eventsource@1.1.2/denonext/eventsource.mjs:2:4187)
    at IncomingMessageForClient.<anonymous> (https://esm.sh/v125/eventsource@1.1.2/denonext/eventsource.mjs:2:3778)
    at IncomingMessageForClient.emit (ext:deno_node/_stream.mjs:1852:9)
    at addChunk (ext:deno_node/_stream.mjs:2874:16)

```

**Comment by denizdogan** [comment]
Just in case anyone finds it useful, I made a Deno version of smee-client, replacing `superagent` with `fetch`, but keeping everything else intact, including outdated and deprecated dependencies...

- https://github.com/denizdogan/smee_client_unofficial
- https://deno.land/x/smee_client_unofficial@v1.2.3

**Comment by Uzlopak** [comment]
should be resolved with v2.0.0
